operators/tokenizer/bpe_kernels.cc

```diff
@@ -641,3 +643,147 @@ OrtxStatus JsonFastTokenizer::Compute(const ortc::Tensor<std::string>& input,
+    
+    std::vector<std::tuple<std::vector<uint8_t>, std::vector<uint8_t>, uint32_t>> local;
+    for (size_t index = 1; index < token.size(); index++) {
+        std::vector<uint8_t> piece_l(token.begin(), token.begin() + index);
+        std::vector<uint8_t> piece_r(token.begin() + index, token.end());
+        if (bpe_ranks.count(piece_l) && bpe_ranks.count(piece_r)) {
+            local.emplace_back(piece_l, piece_r, rank);
+        }
+    }
+    
+    auto compare_bpe_tuples = [&](const std::tuple<std::vector<uint8_t>, std::vector<uint8_t>, uint32_t>& a,
+                                      const std::tuple<std::vector<uint8_t>, std::vector<uint8_t>, uint32_t>& b) {
+      // Compare comparator based on the ranks in bpe_ranks
+      return bpe_ranks[std::get<0>(a)] < bpe_ranks[std::get<0>(b)] ||
+              (bpe_ranks[std::get<0>(a)] == bpe_ranks[std::get<0>(b)] && bpe_ranks[std::get<1>(a)] < bpe_ranks[std::get<1>(b)]);
+    };
+
+    std::sort(local.begin(), local.end(), compare_bpe_tuples);
+    
+    byte_merges.insert(byte_merges.end(), local.begin(), local.end());
+  }
+
+  // Custom comparator that compares the third element of the tuples
+  auto compare_merge_tuples = [](const std::tuple<std::vector<uint8_t>, std::vector<uint8_t>, uint32_t>& a,
+                                      const std::tuple<std::vector<uint8_t>, std::vector<uint8_t>, uint32_t>& b) {
+    return std::get<2>(a) < std::get<2>(b);
+  };
+  
+  std::sort(merges.begin(), merges.end(), compare_merge_tuples);
+
+  // Populate merges
+  for (auto& val : byte_merges) {
+    merges.push_back({TikTokenizer::TokenBytesToString(std::get<0>(val)), TikTokenizer::TokenBytesToString(std::get<1>(val))});
+  }
+
+  const char token_sub[] = "Tokenizer";
+  model_name_ = config.tokenizer_class_.substr(0, config.tokenizer_class_.find(token_sub));
+  json_conf_.name_ = model_name_.c_str();
+  json_conf_.bos_token_ = config.bos_token_.c_str();
+  json_conf_.eos_token_ = config.eos_token_.c_str();
+  json_conf_.unk_token_ = config.unk_token_.c_str();
+  json_conf_.pad_token_ = config.pad_token_.c_str();
+
+  // re-bind the configuration object
+  bpe_conf_ = json_conf_;
+
+  // consider to use SAX parser for large json file
+  nlohmann::json tok_json;
+  ifs >> tok_json;
+  auto model_node = tok_json.find("model");
+  if (model_node == tok_json.end()) {
+    return OrtxStatus(kOrtxErrorCorruptData, "Failed to get model node from tokenizer.json");
+  }
+
+  bbpe_tokenizer_ = std::make_unique<BpeModel>();
+  auto status = bbpe_tokenizer_->Load(vocab,
+                                      merges,
+                                      bpe_conf_.get().GetSpecialTokens().c_str(),
+                                      IsSpmModel(ModelName()));
+
+  if (!status.IsOk()) {
+    return status;
+  }
+
+  add_bos_token_ = config.add_bos_token_;
+  add_eos_token_ = config.add_eos_token_;
+
+  return status;
+}
+
+OrtxStatus TikTokenizer::Compute(const ortc::Tensor<std::string>& input,
```

Comment:
instead of the origin BPE algorithm, can we just use tiktoken tokenize method to compute?

Reply:
i believe after loading the bpe ranks from the .tiktoken file, they also use BPE with a similar algorithm, but i will investigate and improve the compute more after ensuring the loading works firstly.

Reply:
It looks like tiktoken compute equivalent is efficient, https://github.com/openai/tiktoken/blob/c0ba74c238d18b4824c25f3c27fc8698055b9a76/src/lib.rs#L226
here, it worths to have a try.

Reply:
sounds good, will have another PR for this in the future to optimize the initial implementation.